### DISCUSSION TITLE
Windows and demos

### DISCUSSION THREAD

**Discussion by gvanem** [openssl/openssl#24060] [Q&A]
I'm a little surprised to see not all `demos/*.c` sources can be compiled for Windows 
(or rather `_MSC_VER`. I've not tried **MinGW-w64** etc.). 

So with this [patch](https://github.com/openssl/openssl/files/14897167/demos-win32.txt) and my own home-brewed GNU-makefile, I'm able to build these demos for MSVC/clang-cl too:
* `demos/sslecho/sslecho.exe`,
* `demos/bio/saccept.exe` and
* `demos/bio/sconnect.exe`.

(I've not patched the misuse of `errno` for socket errors for `_WIN32`). But is this something you would consider?
And I've tested the above briefly on Windows.

Do not ask me for a PR or a diff to any `demos/*/Makefiles` files; I'm a *n00b* WRT to this. 
*PS*. why are there no `demos/*/build.info` files?

**Comment by nhorman** [comment]
I'd point you to #24047 .

I'm working on fixing this right now :)
